apps/discordbot/bot.py

```diff
@@ -30,6 +39,10 @@ class CSUAClient(discord.Client):
     async def on_ready(self):
         print(f"{self.user} has connected to Discord")
         self.is_phillip = self.user.id == CSUA_PHILBOT_CLIENT_ID
+        csua_bot.announcements_thread = threading.Thread(
+            target=csua_bot.event_announcement, daemon=True
+        )
+        csua_bot.announcements_thread.start()
```

Comment:
The circular dependency makes me a bit uneasy. Haven't completely thought it through, but could `announcements_thread` and `event_announcement` be moved into this class?

Reply:
I originally put this here because we need to wait for `on_ready` to trigger before the `announcement_thread` starts, or else the thread would start too early with no client ready. Do you have something else in mind?

Reply:
Yeah, so I'm wondering why `event_announcement` is a method in `CSUABot` rather than in `CSUAClient`, or even a stand-alone function, considering there's an instance of `CSUAClient` in `CSUABot`. I suppose there isn't really an impact on functionality and it's more about design. My understanding is that `CSUAClient` holds the core parts running the bot, while `CSUABot` is sort of an internal API. Could we get a third opinion?